packages/server/src/jsRunner/tests/isolatedVM.spec.ts

```diff
@@ -0,0 +1,99 @@
+import fs from "fs"
+import path from "path"
+import { IsolatedVM } from "../vm"
+
+function runJSWithIsolatedVM(script: string, context: any) {
+  const runner = new IsolatedVM()
+  return runner.withContext(context, () => {
+    return runner.execute(`(function(){\n${script}\n})();`)
+  })
+}
+
+describe("Test isolated vm directly", () => {
+  it("should handle a very large file", () => {
+    const marked = fs.readFileSync(path.join(__dirname, "marked.txt"), "utf-8")
+    const result = runJSWithIsolatedVM(marked, {
+      trigger: { row: { Message: "dddd" } },
+    })
+    expect(result).toBe("<p>dddd</p>\n")
+  })
+
+  it("handle a mapping case", async () => {
+    const context = {
+      data: {
```

Comment:
What `data.data`?

Reply:
first line of the JS it uses this, its a transformer:
`const dataUnnested = data.data.searchProducts.results`